File: src/main/java/dev/sky_lock/mocar/gui/contents/BaseSideContents.java
package dev.sky_lock.mocar.gui.contents;

import dev.sky_lock.glassy.gui.InventoryMenu;
import dev.sky_lock.glassy.gui.MenuContents;
import dev.sky_lock.glassy.gui.Slot;
import dev.sky_lock.mocar.gui.EditSessions;
import dev.sky_lock.mocar.gui.ModelMenuIndex;
import dev.sky_lock.mocar.item.ItemStackBuilder;
import org.bukkit.Material;
import org.bukkit.entity.Player;
import org.bukkit.inventory.ItemStack;

/**
 * @author sky_lock
 */

public class BaseSideContents extends MenuContents {

    public BaseSideContents(Player player) {
        for (int i = 1; i < 10; i ++) {
            final float baseSide = i * 0.5F;
            ItemStack baseSideItem = ItemStackBuilder.of(Material.LIGHT_BLUE_CONCRETE, 1).name(String.valueOf(baseSide)).build();
            this.addSlot(new Slot(17 + i, baseSideItem, event -> {
                EditSessions.of(player.getUniqueId()).ifPresent(session -> {
                    session.setCollideBaseSide(baseSide);
                });
                InventoryMenu.of(player).ifPresent(menu -> menu.flip(player, ModelMenuIndex.COLLIDE_BOX.value()));
            }));
        }
    }

    @Override
    public void onFlip(InventoryMenu inventoryMenu) {

    }
}

File: src/main/kotlin/dev/sky_lock/pocketlifevehicle/gui/contents/BaseSideContents.kt
package dev.sky_lock.pocketlifevehicle.gui.contents

import dev.sky_lock.menu.InventoryMenu
import dev.sky_lock.menu.InventoryMenu.Companion.of
import dev.sky_lock.menu.MenuContents
import dev.sky_lock.menu.Slot
import dev.sky_lock.pocketlifevehicle.gui.EditSessions.of
import dev.sky_lock.pocketlifevehicle.gui.ModelMenuIndex
import dev.sky_lock.pocketlifevehicle.gui.ModelOption
import dev.sky_lock.pocketlifevehicle.item.ItemStackBuilder
import org.bukkit.Material
import org.bukkit.entity.Player

/**
 * @author sky_lock
 */
class BaseSideContents(player: Player) : MenuContents() {
    override fun onFlip(menu: InventoryMenu) {}

    init {
        for (i in 1..9) {
            val baseSide = i * 0.5f
            val baseSideItem = ItemStackBuilder(Material.LIGHT_BLUE_CONCRETE, 1).name(baseSide.toString()).build()
            addSlot(Slot(17 + i, baseSideItem) {
                of(player.uniqueId).ifPresent { session: ModelOption -> session.collideBaseSide = baseSide }
                of(player).ifPresent { menu: InventoryMenu -> menu.flip(player, ModelMenuIndex.COLLIDE_BOX.ordinal) }
            })
        }
    }
}